### PR TITLE
Arrow_builder_size

### PR DESCRIPTION
**Pull request by jdroenner**
- [ ] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:

<TEXT>

### CODE REVIEW

**Comment by jdroenner** [comment]
[Arrow Builder Byte Size wird nicht richtig berechnet](https://trello.com/c/l6sOToWF/1369-arrow-builder-byte-size-wird-nicht-richtig-berechnet)

File: datatypes/src/collections/feature_collection_builder.rs [hunk 1]
```diff
@@ -286,43 +295,131 @@ where
...
 
         let attributes_size = self
             .builders
-            .values()
-            .map(|builder| {
+            .iter()
+            .map(|(name, builder)| {
                 let values_size = if builder.as_any().is::<Float64Builder>() {
-                    builder.len() * std::mem::size_of::<f64>()
+                    // Float
```
Comment (jdroenner) [inline datatypes/src/collections/feature_collection_builder.rs:325]: this is a lot of duplicated code. Maybe you can use a generic function or macro for that.

Comment (ChristianBeilschmidt) [reply datatypes/src/collections/feature_collection_builder.rs]: [5d17827](https://github.com/geo-engine/geoengine/pull/810/commits/5d17827f1309cd0ac07d6275a807f848e248fdea)

File: datatypes/src/collections/feature_collection_builder.rs [hunk 2]
```diff
@@ -418,3 +476,166 @@ where
         }
     }
 }
+
+fn estimate_primite_arrow_array_size<T: ArrowPrimitiveType>(builder: &dyn ArrayBuilder) -> usize {
```
Comment (jdroenner) [inline datatypes/src/collections/feature_collection_builder.rs:480]: what s a primite? Do you mean primitive?

Comment (ChristianBeilschmidt) [reply datatypes/src/collections/feature_collection_builder.rs]: [f19493f](https://github.com/geo-engine/geoengine/pull/810/commits/f19493f9a55f4fc25cf93bcf5983e0c45cf4d85a)

**Review by ChristianBeilschmidt — APPROVED** [review]

**Review by michaelmattig — APPROVED** [review]

**Pull request merged**